python/python/bystro/proteomics/annotation_interface.py

```diff
@@ -48,93 +42,182 @@ def _flatten(xs: Any) -> list[Any]:  # noqa: ANN401 (`Any` is really correct her
     return sum([_flatten(x) for x in xs], [])
 
 
-def _extract_samples(samples):
-    return [sample[0] for sample in samples]
+def _get_nested_field(data, field_path):
+    """Recursively fetch nested field values using dot notation."""
+    keys = field_path.split(".")
+    value = data
+    for key in keys:
+        try:
+            value = value[key]
+        except (KeyError, TypeError):
+            return None  # Returns None if the field path is not found
+    return value
 
 
-def _get_samples_genes_dosages_from_hit(hit: dict[str, Any]) -> pd.DataFrame:
-    """Given a document hit, return a dataframe of samples, genes and dosages."""
+def _get_samples_genes_dosages_from_hit(
+    hit: dict[str, Any], additional_fields: list[str] | None = None
```

Comment:
With additional fields, can you pull any field you would like or are there any limitations?

Reply:
no limitations; just if you provide fields that are already in the defaults, the defaults would be removed from the additional fields